src/storage-client/src/source/persist_source.rs

```diff
@@ -528,18 +588,270 @@ impl PersistSourceDataStats<'_> {
     }
 }
 
+/// A trait representing a type that can be used in `backpressure`.
+pub trait Backpressureable: Clone + 'static {
```

Comment:
Can we merge this trait with [`MaybeLength`](https://dev.materialize.com/api/rust-private/mz_storage/source/types/trait.MaybeLength.html)?

Reply:
I really do need an accurate size thought, not sure what i'd do with the `None` case...